### DISCUSSION TITLE
Wasmer as a browser as opposed to running wasmer in the browser

### DISCUSSION THREAD

**Discussion by rprovodenko** [wasmerio/wasmer#4319] [Ideas]
The current problem with running wasm in the browser is the overhead to do with the interoperability between wasm and js. Which makes building frontend applications to wasm pointless  for the majority of use cases. I don't think support for DOM or other browser APIs will be added to wasm in the browsers any time soon. Has the wasmer team considered creating a new protocol, e.g. "wasm://...", whereby clicking on a link with such protocol would open wasmer and run whatever wasm application is at that address?

Another option could be forking chromium and running wasmer there, clicking on a "wasm://.." link could open a new tab with the application or even inside the current tab potentially. This would provide for a seamless experience to the user. It would also nudge the web into accepting wasm finally and enable us to write web applications in any language we want, without the performance hits of having to call into js bindings.

**Comment by syrusakbary** [comment]
This is such an interesting idea! I need to sleep a bit more over it, as I think is still early for us, but I need to mature it a bit more
